staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -0,0 +1,336 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bootstrap
+
+import (
+	coordinationv1 "k8s.io/api/coordination/v1"
+	corev1 "k8s.io/api/core/v1"
+	flowcontrol "k8s.io/api/flowcontrol/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	"k8s.io/apiserver/pkg/authentication/user"
+)
+
+// The objects that define an apiserver's initial behavior
+var (
+	PreservedPriorityLevelConfigurations = []*flowcontrol.PriorityLevelConfiguration{
+		DefaultPriorityLevelConfigurationExempt,
+		DefaultPriorityLevelConfigurationDefault,
+	}
+	PreservedFlowSchemas = []*flowcontrol.FlowSchema{
+		DefaultFlowSchemaExempt,
+		DefaultFlowSchemaDefault,
+	}
+)
+
+// Initial PriorityLevelConfiguration objects
+var (
+	DefaultPriorityLevelConfigurationExempt = newPriorityLevelConfiguration(
+		flowcontrol.PriorityLevelConfigurationNameExempt,
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementExempt,
+		},
+	)
+	DefaultPriorityLevelConfigurationDefault = newPriorityLevelConfiguration(
+		"default", // the "catch-all" priority-level
+		flowcontrol.PriorityLevelConfigurationSpec{
+			Type: flowcontrol.PriorityLevelEnablementLimited,
+			Limited: &flowcontrol.LimitedPriorityLevelConfiguration{
+				AssuredConcurrencyShares: 100,
+				LimitResponse: flowcontrol.LimitResponse{
+					Type: flowcontrol.LimitResponseTypeQueue,
+					Queuing: &flowcontrol.QueuingConfiguration{
+						Queues:           128,
+						HandSize:         6,
+						QueueLengthLimit: 100,
+					},
+				},
+			},
+		})
+)
+
+// Initial FlowSchema objects
+var (
+	DefaultFlowSchemaExempt = newFlowSchema(
+		"exempt",
+		flowcontrol.PriorityLevelConfigurationNameExempt,
+		1000, // matchingPrecedence
```

Comment:
the defaulting overwrites zero to sth else so the logically highest is actually 1 i suppose. we cant constraint there's only one unique highest precedence so even a 1-precedence wont always make it the first rule to match.. i think we can constraint the users by clearly document it on godoc?

Reply:
Validation can allow a matching precedence of 1 only for a FlowSchema with the right name.  #85319 does this.

Reply:
Alternatively we could break numeric ties another way; in other places we've gone with alphabetical by name, but I think in this case if we used creationTimestamp with oldest wins, it would do the right thing.

Reserving the e.g. first 100 levels for our own internal use doesn't seem too weird, either.

It is strange to me that "0" means "choose a default" and not the highest priority. We should make the field a pointer so we can tell if a user omitted it and make that an error?